packages/@react-aria/dnd/src/useDrop.ts

```diff
@@ -223,27 +224,41 @@ export function useDrop(options: DropOptions): DropResult {
     e.preventDefault();
     e.stopPropagation();
 
-    if (typeof options.onDrop === 'function') {
-      let dropOperation = DROP_EFFECT_TO_DROP_OPERATION[state.dropEffect];
-      let items = readFromDataTransfer(e.dataTransfer);
-
-      let rect = (e.currentTarget as HTMLElement).getBoundingClientRect();
-      let event: DropEvent = {
-        type: 'drop',
-        x: e.clientX - rect.x,
-        y: e.clientY - rect.y,
-        items,
-        dropOperation
-      };
-
-      // Wait a frame to dispatch the drop event so that we ensure the dragend event fires first.
-      // Otherwise, if onDrop removes the original dragged element from the DOM, dragend will never be fired.
-      // This behavior is consistent across browsers, but see this issue for details:
-      // https://bugzilla.mozilla.org/show_bug.cgi?id=460801
-      setTimeout(() => {
+    let dropOperation = DROP_EFFECT_TO_DROP_OPERATION[state.dropEffect];
+    let items = readFromDataTransfer(e.dataTransfer);
+
+    let rect = (e.currentTarget as HTMLElement).getBoundingClientRect();
+    let event: DropEvent = {
+      type: 'drop',
+      x: e.clientX - rect.x,
+      y: e.clientY - rect.y,
+      items,
+      dropOperation
+    };
+    // Set drop effect in global DnD state for Chrome Android. https://bugs.chromium.org/p/chromium/issues/detail?id=1353951
+    // where onDragEnd always returns "none" as its drop effect.
+    setDropEffect(state.dropEffect);
```

Comment:
Add global onDrop listener to useDrag to get this dropEffect so that dropping on a non-RSP drop target still works in android.

Reply:
Double checked, and it turns out the `state.dropEffect` used here doesn't actually come from onDrop but is set by checking `getDropOperationForPoint` during onDragOver. Since this is the case, I can't move this logic into useDrag since we won't have access to the drop operations of a non-RSP drop target